src/transformers/generation/stopping_criteria.py

```diff
@@ -129,6 +129,27 @@ def __call__(self, input_ids: torch.LongTensor, scores: torch.FloatTensor, **kwa
         return torch.full((input_ids.shape[0],), is_done, device=input_ids.device, dtype=torch.bool)
 
 
+class EOSTokenCriteria(StoppingCriteria):
+    """
+    This class can be used to stop generation whenever the "end-of-sequence" token in generated.
+    By default, it uses the `EOS` token from model's generation config.
+
+    Args:
+        eos_token_id (`Union[int, List[int]]`):
+            The id of the *end-of-sequence* token. Optionally, use a list to set multiple *end-of-sequence* tokens.
```

Comment:
would be nice to also support list of lists ? To have stopping tokens (if the eos is `["<","eos",">"]`)

Reply:
Hmm, is it really true for existing models to have eos tokens as a sequence? I guess you are referring to custom eos tokens, when users want to stop at "<eos>" but they haven't trained the model with "<eos>" as special token. If that's the case, there is `StopStringsCriteria` PR coming or users are free to write their custom criteria

@gante wdyt?

Reply:
Yeah, the case where we want to stop on a string will be covered by #28932 (and is way more complex)

This PR is exclusively to port the existing EOS logic into its own stopping criteria :)

Reply:
Right I forgot it was being covered1